src/components/NewBook.js

```diff
@@ -1,26 +1,50 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+// import PropTypes from 'prop-types';
 import styles from '../styles/NewBook.module.scss';
+import { addBook } from '../redux/books/booksSlice';
 
-const NewBook = () => (
-  <div>
-    <h2>Add New Book</h2>
-    <form className={styles.form}>
-      <input type="text" placeholder="Book title" />
-      <select name="languages" id="lang">
-        <option value="" disabled>
-          Category
-        </option>
-        <option value="javascript">JavaScript</option>
-        <option value="php">PHP</option>
-        <option value="java">Java</option>
-        <option value="golang">Golang</option>
-        <option value="python">Python</option>
-        <option value="c#">C#</option>
-        <option value="C++">C++</option>
-        <option value="erlang">Erlang</option>
-      </select>
-      <input type="submit" value="Add New Book" />
-    </form>
-  </div>
-);
+const NewBook = () => {
+  const val = Math.random() * 1000;
+  const [newBook, setNewBook] = useState({
+    item_id: val.toString(),
+    title: '',
+    author: '',
+    category: 'Nonfiction',
+  });
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(newBook));
+  };
+  return (
+    <div>
+      <h2>Add New Book</h2>
+      <form className={styles.form} onSubmit={handleSubmit}>
+        <input
+          type="text"
+          defaultValue="title"
+          placeholder="Book title"
+          onChange={(e) => setNewBook({ ...newBook, title: e.target.value })}
+        />
+        <select
+          name="author"
+          id="lang"
+          defaultValue="author"
+          onChange={(e) => setNewBook({ ...newBook, author: e.target.value })}
+        >
+          <option value="Biftu Girma">Biftu Girma</option>
+          <option value="Bereket Degu">Bereket Degu</option>
+          <option value="Robert Kiyosaki">Robert Kiyosaki</option>
+          <option value="Marcus R.">Marcus R.</option>
+        </select>
+        <input type="submit" value="Add New Book" />
+      </form>
+    </div>
+  );
+};
 
 export default NewBook;
+
+// NewBook.defaultProps = { books: {} };
+// NewBook.propTypes = { books: PropTypes.arrayOf(PropTypes.string) };
```

Comment:
- [ Required ] I think it would be better if you remove any piece of commented code from your code base. Commented code makes your project unprofessional and is also considered a sign of code that is not clean. Please consider rectifying this.